src/main/scala/org/broadinstitute/dsde/workbench/avram/Avram.scala

```diff
@@ -1,12 +1,17 @@
 package org.broadinstitute.dsde.workbench.avram
 
+import com.typesafe.config.ConfigFactory
+import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+import net.ceedubs.ficus.Ficus._
 import org.apache.commons.dbcp2.BasicDataSource
-import org.broadinstitute.dsde.workbench.avram.config.AvramConfig
+import org.broadinstitute.dsde.workbench.avram.config.{AvramConfig, DbcpDataSourceConfig}
 import org.broadinstitute.dsde.workbench.avram.dataaccess.{HttpSamDao, SamDao}
-import org.broadinstitute.dsde.workbench.avram.util.SlickDatabaseFactory
+import org.broadinstitute.dsde.workbench.avram.db.DbReference
+
+import scala.concurrent.ExecutionContext
 
 /**
-  * Object providing access to all services. This merges configuration and service code to provide
+  * Class providing access to all services. This merges configuration and service code to provide
```

Comment:
Db needs the execution context passed in - so I changed this from an object to a class. Unsure if that's the exact right approach, interested to hear other opinions.

Reply:
Let's explore this further. Changing this to a class means that every instance of a class that extends `BaseEndpoint` will get its own set of these objects, including the database connection pool.

Reply:
I created a PR (#16) that refactors this so it can be an object again. Let me know what you think.